robot_framework/sap/fpo4.py

```diff
@@ -0,0 +1,81 @@
+"""This module handles all actions relating to fpo4."""
+
+import os
+
+from robot_framework.sap.common import set_clipboard, export_file
+
+
+def search_cases(session, fp_list: tuple[str], aftale_list: tuple[str]) -> set[tuple[str]]:
+    """Search for open cases in SAP. Use the fp_list and aftale_list from
+    fplka to filter to reduce the waiting time.
+
+    Args:
+        session: The SAP session object.
+        fp_list: The list of FPs to filter on.
+        aftale_list: The list of aftaler to filter on.
+
+    Returns:
+        set[tuple[str]]: A set of unique fp-aftale pairs (fp, aftale) for open cases.
+    """
+    session.startTransaction("fpo4")
+
+    # Set fp filter
+    session.findById("wnd[0]/usr/tabsTABSTRIP_TAB_B1/tabpTB11/ssub%_SUBSCREEN_TAB_B1:RFKKOP04:1001/btn%_S_GPART_%_APP_%-VALU_PUSH").press()
+    set_clipboard("\r\n".join(fp_list))
+    session.findById("wnd[1]/tbar[0]/btn[24]").press()
+    session.findById("wnd[1]/tbar[0]/btn[8]").press()
+
+    # Set aftale filter
+    session.findById("wnd[0]/usr/tabsTABSTRIP_TAB_B1/tabpTB11/ssub%_SUBSCREEN_TAB_B1:RFKKOP04:1001/btn%_S_VTREF_%_APP_%-VALU_PUSH").press()
+    set_clipboard("\r\n".join(aftale_list))
+    session.findById("wnd[1]/tbar[0]/btn[24]").press()
+    session.findById("wnd[1]/tbar[0]/btn[8]").press()
+
+    # Set outputstyring
+    session.findById("wnd[0]/usr/tabsTABSTRIP_TAB_B1/tabpTB12").select()
+    session.findById("wnd[0]/usr/tabsTABSTRIP_TAB_B1/tabpTB12/ssub%_SUBSCREEN_TAB_B1:RFKKOP04:1002/ctxtP_NM_OPL").text = "STEPHAN"
+
+    # Search
+    session.findById("wnd[0]/tbar[1]/btn[8]").press()
+
+    file_path = os.path.join(os.getcwd(), "fpo4.txt")
```

Comment:
Out of curiosity: Where do the files `fpo4.txt` and `fplka.txt` come from?

Reply:
The robot creates the files as part of the process. "fpo4" and "fplka" are names of SAP transactions. Don't ask me what they mean...
I just decided to call the files the same as the transaction that creates them. They are created and deleted almost instantly, but that had to have some name